hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/HederaStateImpl.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.state.merkle;
+
+import com.hedera.node.app.spi.state.ReadableState;
+import com.hedera.node.app.spi.state.ReadableStates;
+import com.hedera.node.app.spi.state.WritableState;
+import com.hedera.node.app.spi.state.WritableStates;
+import com.hedera.node.app.state.HederaState;
+import com.hedera.node.app.state.impl.InMemoryState;
+import com.hedera.node.app.state.impl.OnDiskState;
+import com.swirlds.common.merkle.MerkleInternal;
+import com.swirlds.common.merkle.impl.PartialNaryMerkleInternal;
+import com.swirlds.common.system.Round;
+import com.swirlds.common.system.SwirldDualState;
+import com.swirlds.common.system.SwirldState2;
+import com.swirlds.common.system.address.AddressBook;
+import com.swirlds.merkle.map.MerkleMap;
+import com.swirlds.virtualmap.VirtualMap;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import org.apache.commons.lang3.NotImplementedException;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+
+public class HederaStateImpl extends PartialNaryMerkleInternal
```

Comment:
This class also needs to override `prehandle()`, right?

Reply:
Yes! And now it does 🥳